torch/_inductor/codecache.py

```diff
@@ -249,35 +253,151 @@ def code_hash(code):
     )
 
 
-def get_code_path(source_code: Union[str, bytes], ext: str, extra: str, binary=False):
-    if binary:
-        source_code = repr(source_code)
-    basename = code_hash(source_code + extra)
+def serialize_fx_arg(arg: Any):
```

Comment:
is this the right name? 

For the guard pickling part why not define a `__setstate__/__getstate__` for `torch/_guards/Guard`

Reply:
changed the name to `serialize_compiled_fx_arg`

for the guards, I think the idea was to only save the relevant parts from the example_inputs